src/diagram/components/ui/expandable-input.tsx

```diff
@@ -70,6 +78,7 @@ export const ExpandableInput = ({
           autoComplete="off"
           className={inputClasses}
           data-testid={`variable-${title}`}
+          disabled={disabled}
```

Comment:
I don't think you need to pass disabled here because you already check to make sure disabled is false before you try to render a `NumberInput`.

Reply:
I'm pretty sure it's needed because the check for `!disabled` is part of a trio of conditions that have to be met to render a number input. Even if the input type is a string, it may be disabled or not disabled.

Reply:
Oh, I think maybe you need to expand the code in this view to see the entire `if...else`? It doesn't pass `disabled` to `NumberInput`, only to the `textarea` in the `else` part.